code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -23,10 +27,19 @@
 	owner.special_role = special_role
 	if(give_objectives)
 		forge_traitor_objectives()
+	create_contracts()
 	finalize_traitor()
 	RegisterSignal(owner.current, COMSIG_MOVABLE_HEAR, .proc/handle_hearing)
 	..()
 
+/datum/antagonist/traitor/proc/create_contracts()
```

Comment:
You could probably delay generating these until  they first access the contract interface.